### PR TITLE
Create database based on diagram

### PR DESCRIPTION
**Pull request by dieum512**
In this milestone we:

- Created the database based on the diagram and
- Added indexes to the Foreign keys
- Tables as bellow: patients, medical_history,  invoices, treatments, invoice_items

Thank you for the time and energy to review this work

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+-- create table invoices 
+CREATE TABLE invoices(id INTEGER PRIMARY KEY, total_amount DECIMAL, generated_at TIMESTAMP, payed_at TIMESTAMP, medical_history_id INTEGER REFERENCES medical_histories(id)); 
+
+--create table treatments
+CREATE TABLE treatments(id INTEGER  medical_histories(id), type VARCHAR(255), name VARCHAR(255));
+
+-- create table invoice_items
+CREATE TABLE invoice_items(id INTEGER PRIMARY KEY, unit_price DECIMAL, quantity INTEGER, total_price DECIMAL, invoice_id INTEGER, treatment_id INTEGER REFERENCES treatments(id));
+
```
Comment (topeogunleye) [inline schema_based_on_diagram.sql:26]: - As written in the project requirements, I kindly suggest that all "join table" for the many-to-many relastionships is created, even if they are not showed on the diagram. Therefore you should kindly create a table between `medical_histories` and `treatments` (name can be different):


```.sql
-- Create join table medical_histories_has_treatments
CREATE TABLE medical_histories_has_treatments(
  medical_history_id INTEGER REFERENCES medical_histories(id),
  treatment_id INTEGER REFERENCES treatments(id),
  PRIMARY KEY (medical_history_id, treatment_id)
);
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+-- create indexes
+CREATE INDEX ON medical_histories(patient_id);
+CREATE INDEX ON invoices(medical_history_id);
+CREATE INDEX ON invoice_items(invoice_id);
+CREATE INDEX ON invoice_items(treatment_id);
```
Comment (topeogunleye) [inline schema_based_on_diagram.sql]: - [ ] After you create a table between `medical_histories` and `treatments` for the many-to-many relastionship between them. Remember to have it's foreign keys indexed. for example:

```.sql
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`
```

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @MMhaidari, @dieum512, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
- Implemented all required tables with correct columns and data types. ✔️
- Set primary keys for all tables. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ]  Next time, and for future commits, I kindly suggest that you please make use of imperative mood in the subject line for your commit messages.
:x: Bad Commit Messages :x:
image

### ✔️ Good Commit Messages: ✔️
✔️  Add fix for dark mode toggle state
✔️  Update README file.
✔️  Add styles to h1.

- [ ] Please refer to this article: please read it and apply it in your work from now on: [Professional repo - basic rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md);

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MMhaidari** [comment]
# What is added new to the project
- [ ] many to many relation added to the tables as instructed
- [ ] indexs also added to the foreign keys

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

**Hi @Team,** 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented requested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "bug fixed".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**